src/System.Slices/src/System/Contract.cs

```diff
@@ -41,6 +41,18 @@ public static void RequiresInInclusiveRange(int start, int length)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void RequiresInInclusiveRange(int start, int length, int existingLength)
+        {
+            if ((uint)start > (uint)existingLength
+                || length < 0
+                || (uint)(start + length) > (uint)existingLength)
+            {
+                throw NewArgumentOutOfRangeException();
+            }
+        }
+
+        
```

Comment:
In your case, it's length, but I worry that people will reuse the very convenient method (especially once we make the contract class public, which I want to do), and they will use it with something that is not guaranteed to be positive.